ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java

```diff
@@ -282,14 +282,17 @@ private void loadLogSegments(long lastIndexInSnapshot,
 
   @Override
   public LogEntryProto get(long index) throws RaftLogIOException {
-    final ReferenceCountedObject<LogEntryProto> ref = retainLog(index);
-    if (ref == null) {
-      return null;
-    }
+    ReferenceCountedObject<LogEntryProto> ref = null;
     try {
+      ref = retainLog(index);
```

Comment:
Similar to the previous case, `retainLog(index)` should be called before the try-block.